ublox-bluetooth-commands.go

```diff
@@ -171,12 +171,28 @@ func (ub *UbloxBluetooth) discoveryCommandWithContext(ctx context.Context, scant
 // ConnectToDevice attempts to connect to the device with the specified address.
 func (ub *UbloxBluetooth) ConnectToDevice(address string, onConnect DeviceEvent, onDisconnect DeviceEvent) (err error) {
 	var d []byte
+	var cr *ConnectionReply
+
 	d, err = ub.writeAndWait(ConnectCommand(address), true)
 	if err != nil {
 		return err
 	}
 
-	var cr *ConnectionReply
+	// Flush out disconnect request - Bug in current Nina firmware.
+	err = ub.ATCommand()
+	if err == nil {
+		if err == ErrUnexpectedDisconnect {
+			// Try again...
```

Comment:
Do i need to read from the serial port here assuming there response from the AT command will be queued after the disconnect reply?

Reply:
The serial port itself should underneath the EDM driver.

The EDM driver should multiplex/demultiplex the different streams from teh NINA board into the constituent streams which include at least:

1) [To NINA] AT commands
2) [From NINA] AT responses
3) [From NINA] Connect indications (+UUBTACLC)
4) [From NINA] Disconnect indications (+UUBTACLD)

So when you send an AT Command you should expect to receive an AT response on the AT Channel. The connect/disconnect messages will apaper 'out of band' on other channels.